packages/admin/src/Resources/RelationManagers/Concerns/CanAttachRecords.php

```diff
@@ -92,7 +93,10 @@ protected static function getAttachFormRecordSelect(): Select
                     ->whereDoesntHave($livewire->getInverseRelationshipName(), function (Builder $query) use ($livewire): void {
                         $query->where($livewire->ownerRecord->getQualifiedKeyName(), $livewire->ownerRecord->getKey());
                     })
-                    ->pluck($displayColumnName, $relationship->getRelated()->getKeyName())
+                    ->get()
+                    ->mapWithKeys(static fn (Model $record) => [
+                        $record->{$relationship->getRelated()->getKeyName()} => static::getRecordTitle($record),
```

Comment:
Here and on line 121, we can store the result of `$relationship->getRelated()->getKeyName()` in a variable so that we don't call it more than once.

Reply:
That's 2 different methods. We would need to extract `$relationship` and the new variable outside the `Select` component scope then.

Reply:
I meant two variables, one in each closure.

Reply:
Ah, now I get it. We have a loop with `mapWithKeys()` 😅